### PR TITLE
sendpayment / tumbler: dynamic tx fee estimation

### PR DESCRIPTION
**Pull request by AlexCato**
If the user does not give an initial fee estimation, dynamically estimate the initial fee from the blockchain. Required for non-sweeping sendpayments.

This helps to fix issue #544 . Discussion also in PR #550

### CODE REVIEW

**Comment by AdamISZ** [comment]
After the long discussion in #550 , I think this is fine. (annoying we are still not getting Travis build triggers on PRs). Apologies for misunderstanding it at first but I guess it's good we're talking about this rather hairy topic again!

If no more discussion I'll merge it shortly.

File: sendpayment.py [hunk 1]
```diff
@@ -295,6 +295,13 @@ def main():
     else:  # choose randomly (weighted)
         chooseOrdersFunc = weighted_order_choose
 
+    # Dynamically estimate a realistic fee if it currently is the default value.
+    # At this point we do not know even the number of our own inputs, so
+    # we guess conservatively with 2 inputs and 2 outputs each
+    if options.txfee == 5000:
```
Comment (chris-belcher) [inline sendpayment.py]: I'd say make the default txfee 0 instead of 5000, to mean estimate it.

**Comment by AlexCato** [comment]
Agreed, belcher. Will use -1 as default.

Also this needs one more modification, otherwise the now approximately accurate estimates are still doubled later in the code which might lead to more often "insufficient funds" errors.

**Comment by AlexCato** [comment]
Would like to remove the multiplication by 2 from this line, so only an 100% increase over the first estimation is allowed instead of 200% increase (and would change the same thing in tumbler.py):

https://github.com/AlexCato/joinmarket/blob/ad4efdbc4254746bcc220e84e6f47f45bb646823/sendpayment.py#L112

Reason: less insufficient funds errors, also the estimates should be good enough to not need that much margin for error

Any objections?

**Comment by AdamISZ** [comment]
@AlexCato yes probably that makes sense. It's more an art than a science at this point; but extra 1x (so 2x in total) feels about right to me.

**Comment by AlexCato** [comment]
Updated the PR with belcher's default and the slightly decreased fee estimation discussed.

**Comment by chris-belcher** [comment]
I think it looks good.
So before we merge this you should squish your commits together into one commit.

Here is the tutorials I used in case it helps
http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html

**Pull request merged**